lib/src/page/page.dart

```diff
@@ -185,8 +185,12 @@ class Page {
     await page._initialize();
 
     if (viewport != null) {
-      await page.setViewport(viewport);
-    }
+      try {
+        await page.setViewport(viewport);
+      } catch (e) {
+        _logger.warning('Unable to set viewport for ${target.url}: $e');
```

Comment:
Note: This was added because calling `setViewport` on the Chrome DevTools target (after being set to `type` "page") throws the following error: "Command can only be executed on top-level targets"